docs/src/docs/usage/faq.mdx

```diff
@@ -35,15 +35,22 @@ Usually this options is used during development on local machine and compilation
 
 ## Why do you have `typecheck` errors?
 
-`typecheck` is like the front-end of a Go compiler, parses and type-checks Go code, it manages compilation errors.
+`typecheck` is like a front-end for the Go compiler errors.
+Compilation errors are identified/labeled as reports of `typecheck` but they are not produced by a linter called `typecheck`.
+
+`typecheck` is not a linter, it doesn't perform any analysis,
+it's just a way to identify, parse, and display compiling errors (produced by the `types.Checker`) and some linter errors.
 
 It cannot be disabled because of that.
 
 Of course, this is just as good as the compiler itself and a lot of compilation issues will not properly show where in the code your error lies.
 
-`typecheck` is not a real linter, it's just a way to parse compiling errors (produced by the `types.Checker`) and some linter errors.
+As consequence, the code to analyze should compile.
+It means that if you try to run an analysis on a single file or a group of files or a package or a group of packages,
+with dependencies on other files or packages of your project, as it doesn't compile (because of the missing pieces of code),
+it also cannot be analyzed.
```

Comment:
Should a short version of this be added to the troubleshooting list to clarify that not only try `go build ./...` but also try `golangci-lint run ./...`?

Reply:
It's complex to summarize this sentence because each part is important to avoid ambiguity.